Automattic/jetpack
# Button block: check block root parent attributes

When passing through attributes to a child Button block using `passthroughAttributes`, we want to make sure the Button looks to the correct parent block to fetch the attributes.

If a Calendly block is nested inside another block, previously the Button was looking to the top of the nesting tree rather than its parent.

## Changes proposed in this Pull Request

Rather than use [getBlockHierarchyRootClientId](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#getBlockHierarchyRootClientId), looks at the top of the tree in which a block is nested, we look to [getBlockRootClientId](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#getBlockRootClientId) so that we can get the immediate parent block.

<img width="1059" alt="Screen Shot 2021-05-19 at 3 32 31 pm" src="https://user-images.githubusercontent.com/6458278/118761253-6ea89500-b8b7-11eb-9437-0921fa31c0c1.png">

Fixes https://github.com/Automattic/jetpack/issues/17425
Fixes https://github.com/Automattic/wp-calypso/issues/52687

## Jetpack product discussion
See bug reports: 

https://github.com/Automattic/jetpack/issues/17425
https://github.com/Automattic/wp-calypso/issues/52687

#### Does this pull request change what data or activity we track or use?
Nope.

## Testing instructions:
1. Insert a Calendly block into the editor, and another inside a group or column block.
2. Add a Calendly url (you might have to sign up to Calendly) to each. Change each button to the "Link" style so that the button appears.
3. Change the buttons' text color.
4. Save and check the frontend.

- [ ] The button link should open the calendar.
- [ ] The text color should correspond to the one you chose.
- [ ] The universe is enormous. Feeling small is normal.
